### PR TITLE
Improved charts, Added provenance tab

### LINKED ISSUE #26: Display information on provenance when available
The `overview` API endpoint returns a JSON containing a `datasets` property with the following fields:
```
"title", "description", "contact_info",
"dates", "stored_in", "spatial_coverage",
"types", "privacy", "distributions",
"dimensions", "primary_publications", "citations",
"produced_by", "creators", "licenses",
"acknowledges", "keywords", "version",
"extra_properties"
```

TODO: display these information as a text that will be written as the footer of the overview tab.

Note that several properties might have an empty value. As a first step, display a list of key/value pairs and ignore empty values. It's possible that in the future, a list of compulsory fields might be enforced programmatically. Don't implement this yet but make the code flexible/simple enough to allow this, if that's not too much overhead.

Note that `extra_properties` is a nested field that should probably be "un-nested".

### PR DESCRIPTION
**Pull request by SanjeevLakhwani**
## Improved data display for "Missing" field in bar charts/ histograms
- Coming from the issue that in a lot of data, missing field dominated the charts requiring large y-axis scales and harder to understand data, due to this now missing field would not be shown as a column but as a descriptive text with the chart card
- For more clarity it contains an person logo attached to the next but for future we would need to differentiate between individuals or experiments

## Added "Other" Section to pie charts 
- In really diversely distributed data, there existed really small sections which are hard to focus on, therefore created an Other section which combines all sections below a specific sector size, given that there are more than one such sectors

## Missing Sector in Pie chart now has a common color scheme
- Every Sector with key "Missing" (case insensitive) would now have a grey color

## Added a Provenance tab
- This Tab is to display information from where the data comes from
- It displays information given by katsu, which is the provenance for all the dataset held in katsu
- It resolves #26 
- It relies on `/api/public_dataset` for response. if the response contains a dats_file it renders that otherwise the parent object

## Added a max limit to number of charts displayed on first render
- Now you can limit the number of charts that would be displayed
  - Note: Local storage still overwrites whether a chart is displayed or not, so it is only valid when local storage isn't present
- To test with less number of charts, the default max being 9, change the variable `MAX_CHARTS` in configConstants.js to desired number and don't forget to clear localstorage!

### CODE REVIEW

**Review by ppillot — COMMENTED** [review]
I've made a cursory look through the changes. I've noticed that there is a difference with bento_web in the way the "other" category feature is implemented:
- in bento_web the special case where **only one category** is bellow the threshold is detected so that no "Other" is displayed
- in this commit I don't think that the code makes this distinction

**Comment by SanjeevLakhwani** [comment]
> I've made a cursory look through the changes. I've noticed that there is a difference with bento_web in the way the "other" category feature is implemented:
> 
> * in bento_web the special case where **only one category** is bellow the threshold is detected so that no "Other" is displayed
> * in this commit I don't think that the code makes this distinction

Thanks for pointing it out, I have made the change and also added another change for missing sector in pie charts. I have added details in the PR statement

File: src/js/utils/string.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const toTitleCase = (str) =>
+  str.replace(/\w\S*/g, (txt) => txt.charAt(0).toUpperCase() + txt.substring(1).toLowerCase());
```
Comment (ppillot) [inline src/js/utils/string.js:2]: Not sure it is used again, but if you ever need to do this, consider the CSS alternative `text-transform: capitalize`

File: src/js/components/overview/MakeChartCard.js [hunk 2]
```diff
@@ -29,6 +30,15 @@ const MakeChartCard = ({ section, chart, onRemoveChart }) => {
...
   ));
 
+  const missingCount = data.find((e) => e.x === 'missing')?.y ?? 0;
+  const missingDataText = (
+    <Typography.Text key={-1} type="secondary" italic>
+      <TeamOutlined /> {missingCount} missing
+    </Typography.Text>
+  );
+
+  if (missingCount) ed.unshift(missingDataText);
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/overview/MakeChartCard.js]: done

File: src/js/components/overview/charts/BentoPie.js [hunk 3]
```diff
@@ -36,7 +36,18 @@ const labelShortName = (name) => {
 function BentoPie({ data, height }) {
   const [activeIndex, setActiveIndex] = useState(undefined);
 
-  data = data.filter((e) => e.y !== 0);
+  // Changing immutable data to mutable data
+  data = [...data];
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/overview/charts/BentoPie.js:40]: Merci

File: src/js/components/overview/charts/BentoPie.js [hunk 4]
```diff
@@ -36,7 +36,18 @@ const labelShortName = (name) => {
...
+
+  // combining sections with less than OTHER_THRESHOLD
+  const sum = data.reduce((acc, e) => acc + e.y, 0);
+  const length = data.length;
+  const temp = data.filter((e) => e.y / sum > OTHER_THRESHOLD);
+  data = temp.length === length - 1 ? data : temp;
+  if (data.length !== length) {
+    data.push({ x: 'Other', y: sum - data.reduce((acc, e) => acc + e.y, 0) });
+  }
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/overview/charts/BentoPie.js]: done

File: src/js/components/Util/LinkIfUrl.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Typography } from 'antd';
+
+// Renders text as link if the text provided is a valid url.
+const LinkIfUrl = ({ text }) => {
+  if (text.match(/^(http|https):\/\/[^ "]+$/)) {
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/Util/LinkIfUrl.js]: done

File: src/js/components/Provenance/Tables/IsAboutTable.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+const IsAboutTable = ({ isAbout }) => {
+  return (
+    <Table
+      dataSource={isAbout}
+      columns={[
+        { title: 'Name', dataIndex: 'name', key: 'name' },
+        {
+          title: 'Identifier',
+          dataIndex: 'identifier.identifier',
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/Provenance/Tables/IsAboutTable.js:13]: left it as that for the reader to get the path but antd cannot access nested properties

File: src/js/utils/string.js (hunk 1 above)
Comment (SanjeevLakhwani) [reply src/js/utils/string.js]: It is not used, therefore removed it. It was previously used for the older version of provenance but my bad missed it in the cleanup.

File: src/js/components/overview/charts/BentoPie.js [hunk 7]
```diff
@@ -36,7 +36,18 @@ const labelShortName = (name) => {
...
+
   const bentoFormatData = data.map((e) => ({ name: e.x, value: e.y }));
   const totalCount = bentoFormatData.reduce((sum, e) => sum + e.value, 0);
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/overview/charts/BentoPie.js]: It is I changed it

**Review by ppillot — APPROVED** [review]

**Comment by ppillot** [comment]
Thanks for all these changes!
The API error reporting function is a nice addition too.
I've added some minor modification to handle the shape of the response coming from Katsu WRT datasets properties, and did some corrections to some comments.
Tested and approved!

**Pull request merged**